var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -368,35 +380,43 @@ def patch(self):
 
     @property
     def common_config_args(self):
+        spec = self.spec
+        version = self.version
+
         # incomplete list is here http://doc.qt.io/qt-5/configure-options.html
         config_args = [
             '-prefix', self.prefix,
             '-v',
             '-opensource',
-            '-{0}opengl'.format('' if '+opengl' in self.spec else 'no-'),
-            '-{0}'.format('debug' if '+debug' in self.spec else 'release'),
+            '-{0}opengl'.format('' if '+opengl' in spec else 'no-'),
+            '-{0}'.format('debug' if '+debug' in spec else 'release'),
             '-confirm-license',
             '-optimized-qmake',
             '-no-pch',
         ]
 
-        if self.spec.variants['freetype'].value == 'spack':
-            config_args.extend([
-                '-system-freetype'
-            ])
-            config_args.extend(
-                self.spec['freetype'].headers.include_flags.split()
-            )
+        def use_spack_dep(spack_pkg, qt_name=None):
+            pkg = spec[spack_pkg]
+            config_args.append('-system-' + (qt_name or spack_pkg))
+            if not pkg.external:
+                config_args.extend([
+                    pkg.libs.search_flags,
+                    pkg.headers.include_flags
+                ])
```

Comment:
Will this work if `search_flags` or `include_flags` include multiple flags? I think that's why we had `include_flags.split()` above.

Reply:
I didn't notice the `split()` honestly -- the other repetitions of this code didn't have it. I will add it just in case.